sdk/core/azure-core/inc/azure/core/operation.hpp

```diff
@@ -30,7 +30,6 @@ namespace Azure { namespace Core {
     virtual std::unique_ptr<Http::RawResponse> PollInternal(Context const& context) = 0;
     virtual Response<T> PollUntilDoneInternal(std::chrono::milliseconds period, Context& context)
         = 0;
-    virtual Azure::Core::Http::RawResponse const& GetRawResponseInternal() const = 0;
```

Comment:
I believe the intent of this virtual method was to guard against `null` and prevent a seg-fault in derived classes which implement `Operation<T>`. It was added in the following PR by @vhvb1989.
See this test case:
https://github.com/Azure/azure-sdk-for-cpp/pull/1875#discussion_r592868084

I think `GetRawResponse()` needs to call it in `operation.hpp` similar to how `Poll` calls `PollInternal`, and we may have missed that detail in the PR that introduced this method.

Reply:
Looking back at how we ended up here, we had initially designed `Operation<T>` to have a pure **virtual** `GetRawResponse()` method that all derived classes would implement.

The PR that introduced it doesn't mention the design rationale explaining why we want that: https://github.com/Azure/azure-sdk-for-cpp/pull/1803
> Added the pure virtual method

I don't know the reason for that initial design, maybe it was to allow the derived classes to special case how to get the raw response, and maybe do some pre-processing before returning. It looks like we no longer want `GetRawResponse()` to be virtual at all (even via GetRawResponseInternal).

Since this is private inheritance, we can always add that extension point in the future, as needed.

The current `GetRawResponse()` **instance** method does the appropriate null checking, so the seg-fault error doesn't show up anymore.

**I don't see any ownership transfer issues with `m_rawResponse` now, at least not with `GetRawResponse()`, since `Operation<T>` owns it.**